graal/backends/core/analyzers/cloc.py

```diff
@@ -32,45 +32,99 @@ class Cloc(Analyzer):
+        for line in message.strip().split("\n"):
             if flag:
                 if not line.startswith("-----"):
                     digested = " ".join(line.split())
                     info_file = digested.split(" ")
                     blank_lines = int(info_file[2])
                     commented_lines = int(info_file[3])
                     loc = int(info_file[4])
+                    results["blanks"] = blank_lines
+                    results["comments"] = commented_lines
+                    results["loc"] = loc
+                    break
 
-                    result['blanks'] = blank_lines
-                    result['comments'] = commented_lines
-                    result['loc'] = loc
+            if line.lower().startswith("language"):
+                flag = True
+
+        return results
+
+    def __analyze_repository(self, message):
+        """Add information LOC, Total files, blank and commented lines using CLOC
+
+        :param message:
+
+        :returns result: dict of the results of the analysis over a repository
+        """
+
+        results = {}
+        flag = False
+
+        for line in message.strip().split("\n"):
+            if flag:
+                if line.lower().startswith("sum"):
                     break
+                elif not line.startswith("-----"):
+                    digested = " ".join(line.split())
+                    info_file = digested.split(" ")
+                    blank_lines = int(info_file[2])
+                    commented_lines = int(info_file[3])
+                    loc = int(info_file[4])
+                    language = info_file[0]
+                    language_result = {
+                        "total_files": int(info_file[1]),
+                        "blanks": blank_lines,
+                        "comments": commented_lines,
+                        "loc": loc
+                    }
+                    results[language] = language_result
 
             if line.lower().startswith("language"):
                 flag = True
 
-        result['ext'] = file_path.split(".")[-1]
-        return result
+        return results
+
+    def analyze(self, **kwargs):
+        """Add information using CLOC
+
+        :param file_path: file path
+        :param repository_level: set to True if analysis has to be performed on a repository
+
+        :returns result: dict of the results of the analysis
+        """
+
+        file_path = kwargs['file_path']
+        repository_level = kwargs.get('repository_level', False)
+
+        try:
+            message = subprocess.check_output(['cloc', file_path]).decode("utf-8")
+        except subprocess.CalledProcessError as e:
+            raise GraalError(cause="Cloc failed at %s, %s" % (file_path, e.output.decode("utf-8")))
+        finally:
+            subprocess._cleanup()
+
+        if repository_level:
+            results = self.__analyze_repository(message)
+            results['ext'] = file_path.split(".")[-1]
```

Comment:
GraalRepository provides a static method for this: https://github.com/chaoss/grimoirelab-graal/blob/master/graal/graal.py#L417

Is it possible to replace it ?

Reply:
Yes. 
Thanks for the suggestion :)